### ISSUE TITLE
[2.107.0] Uploading metadata to iTunes Connect error: source sequence is illegal/malformed utf-8

### ISSUE THREAD

**Issue by SEnglishSMP** [fastlane/fastlane#13608]
This is a re-entry of issue #13125, which shouldn't have been auto-closed since it was never resolved.

### New Issue Checklist
- [x] Updated fastlane to the latest version
- [x] I read the Contribution Guidelines
- [x] I read docs.fastlane.tools
- [x] I searched for existing GitHub issues

### Issue Description
I'm getting an error while running fastlane deliver

/Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/base.rb:66:in `to_json': [!] source sequence is illegal/malformed utf-8 (JSON::GeneratorError)

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>fastlane deliver --force -u *** -a com.gosmp.smpmobile -i /Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/output/iphoneos/Distribution/_XcodeTaskExport_SMPNew/SMPNew.ipa --skip_binary_upload true -m /Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/AppStore/metadata --skip_screenshots true --submit_for_review true
[21:26:24]: [33mGet started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
[21:26:25]: Sending anonymous analytics information
[21:26:25]: Learn more at https://docs.fastlane.tools/#metrics
[21:26:25]: No personal or sensitive data is sent.
[21:26:25]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[21:26:25]: Login to App Store Connect (***)
[21:26:27]: Login successful

+--------------------------------------+-----------------------------------------------------------------------------------------------------------------------+
|                                                                   deliver 2.107.0 Summary                                                                    |
+--------------------------------------+-----------------------------------------------------------------------------------------------------------------------+
| force                                | true                                                                                                                  |
| username                             | ***                                                                                                     |
| app_identifier                       | com.gosmp.smpmobile                                                                                                   |
| ipa                                  | /Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/output/iphoneos/Distribution/_XcodeTaskExport_SMPNew/SMPNew.ipa |
| skip_binary_upload                   | true                                                                                                                  |
| metadata_path                        | /Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/AppStore/metadata                                               |
| skip_screenshots                     | true                                                                                                                  |
| submit_for_review                    | true                                                                                                                  |
| screenshots_path                     | ./screenshots                                                                                                         |
| app_version                          | 1.7.11026                                                                                                             |
| platform                             | ios                                                                                                                   |
| edit_live                            | false                                                                                                                 |
| use_live_version                     | false                                                                                                                 |
| skip_metadata                        | false                                                                                                                 |
| skip_app_version_update              | false                                                                                                                 |
| overwrite_screenshots                | false                                                                                                                 |
| reject_if_possible                   | false                                                                                                                 |
| automatic_release                    | false                                                                                                                 |
| phased_release                       | false                                                                                                                 |
| run_precheck_before_submit           | true                                                                                                                  |
| precheck_default_rule_level          | warn                                                                                                                  |
| ignore_language_directory_validation | false                                                                                                                 |
| precheck_include_in_app_purchases    | true                                                                                                                  |
+--------------------------------------+-----------------------------------------------------------------------------------------------------------------------+

[21:26:28]: Making sure the latest version on App Store Connect matches '1.7.11026' from the ipa file...
[21:26:30]: Successfully set the version to '1.7.11026'
[21:26:31]: Loading '/Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/AppStore/metadata/en-US/description.txt'...
[21:26:31]: Loading '/Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/AppStore/metadata/en-US/release_notes.txt'...
[21:26:31]: Loading '/Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/AppStore/metadata/en-US/support_url.txt'...
[21:26:31]: Loading '/Users/vsts/agent/2.141.1/work/r1/a/_iOS release/drop/AppStore/metadata/copyright.txt'...
[21:26:32]: Activating language en-US...
[21:26:57]: Uploading metadata to App Store Connect

Looking for related GitHub issues on fastlane/fastlane...

➡️  Uploading metadata to iTunes Connect error: source sequence is illegal/malformed utf-8 
    https://github.com/fastlane/fastlane/issues/13125 [closed] 4 💬
    5 weeks ago

➡️  Uploading metadata to iTunes Connect error: source sequence is illegal/malformed utf-8
    https://github.com/fastlane/fastlane/issues/11961 [closed] 3 💬
    08 Jun 2018

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/base.rb:66:in `to_json': [!] source sequence is illegal/malformed utf-8 (JSON::GeneratorError)
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/base.rb:66:in `to_json'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/tunes/tunes_client.rb:432:in `block (2 levels) in update_app_version!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/connection.rb:384:in `block in run_request'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/connection.rb:398:in `block in build_request'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/request.rb:26:in `block in create'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/request.rb:25:in `tap'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/request.rb:25:in `create'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/connection.rb:394:in `build_request'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/connection.rb:379:in `run_request'
	from /usr/local/lib/ruby/gems/2.5.0/gems/faraday-0.15.3/lib/faraday/connection.rb:175:in `post'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:701:in `block in send_request'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:533:in `with_retry'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:700:in `send_request'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:586:in `request'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/tunes/tunes_client.rb:430:in `block in update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/tunes/tunes_client.rb:1433:in `with_tunes_retry'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/tunes/tunes_client.rb:429:in `update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/spaceship/lib/spaceship/tunes/app_version.rb:403:in `save!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/deliver/lib/deliver/upload_metadata.rb:130:in `upload'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/deliver/lib/deliver/runner.rb:120:in `upload_metadata'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/deliver/lib/deliver/runner.rb:38:in `run'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/deliver/lib/deliver/commands_generator.rb:79:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/deliver/lib/deliver/commands_generator.rb:182:in `run'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/deliver/lib/deliver/commands_generator.rb:17:in `start'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/fastlane/lib/fastlane/cli_tools_distributor.rb:100:in `take_off'
	from /Users/vsts/.gem-cache/gems/fastlane-2.107.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
</pre>
</details>

### Environment

We're using a Visual Studio Team Services hosted build server. I can't run `fastlane env`, but this is what Microsoft says is installed on the current machine image.

<details>
  <pre>-- Installation software information ---------

ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]
Python 2.7.10
Python 3.7.0
pip 18.0 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)
Node.js v6.14.4
NPM 3.10.10
Git: git version 2.19.0
Git LFS: git-lfs/2.5.2 (GitHub; darwin amd64; go 1.11)
curl 7.54.0 (x86_64-apple-darwin17.0) libcurl/7.54.0 LibreSSL/2.0.20 zlib/1.2.11 nghttp2/1.24.0
GNU Wget 1.19.5 built on darwin17.5.0.
Homebrew 1.7.6
xctool 0.3.4
Bundler version 1.16.1
CocoaPods 1.5.3
xcpretty 0.3.0
Nomad CLI 2.7.6
Nomad CLI ipa 0.14.3
Mobile Center CLI 0.12.0
App Center CLI 1.1.2
fastlane 2.105.2
Carthage 0.30.1
Gradle 4.10.2
PowerShell 6.1.0
Yarn 1.10.1
java 1.7 jdk: 1.7.0_80
java 1.8 jdk: 1.8.0_181 (default)
java 9 jdk: 9.0.4
java 10 jdk: 10.0.2
java 11 jdk: 11
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z)
.NET SDK 1.0.1 1.0.4 1.1.10 1.1.4 1.1.5 1.1.7 1.1.8 1.1.9 2.0.0 2.0.3 2.1.100 2.1.101 2.1.102 2.1.103 2.1.104 2.1.105 2.1.2 2.1.200 2.1.201 2.1.202 2.1.300 2.1.301 2.1.302 2.1.4 2.1.400 2.1.401 2.1.402
azure-cli (2.0.46)
Go 1.11
Subversion (SVN) 1.10.2
Miniconda 4.5.11
macOS toolcache Python 2.7.15 3.4.8 3.5.5 3.6.5 3.7.0

-- OS X info ----------------------------------

sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G65

system_profiler SPSoftwareDataType
Software:

System Software Overview:

System Version: macOS 10.13.6 (17G65)
Kernel Version: Darwin 17.7.0
Boot Volume: Macintosh HD
Boot Mode: Normal
Computer Name: Mac (215)
User Name: vsts (vsts)
Secure Virtual Memory: Enabled
System Integrity Protection: Enabled
Time since boot: 2:29

-- Disk ---------------------------------------

Total size: 407162134528 bytes (379Gi)
Free: 111746826240 bytes (104Gi)</pre>
</details>

**Comment by SEnglishSMP** [comment]
As stated, can't include the output of `fastlane env`.

**Comment by brightsider** [comment]
Same issue. How to upload?

**Comment by eXhausted** [comment]
@brightsider  seems it's 2.109.0 version's issue, try to roll back to 2.108.0

**Comment by Skoti** [comment]
I am experiencing the same issue. It works in 2.108.0 though so it is a regression.

Same error but in a different line:
`{redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/client.rb:783:in `to_json': [!] source sequence is illegal/malformed utf-8 (JSON::GeneratorError)`

I tried submitting an app for review using deliver but after uploading app icon step it crashes with the error above.

parameters I used for deliver:
```
  skip_app_version_update:true
  skip_binary_upload:true
  skip_metadata:true
  skip_screenshots:true
  submit_for_review:true
  update_submission_information:true
```

My output:
<details>
  <pre>
[12:12:20]: ---------------------
[12:12:20]: --- Step: deliver ---
[12:12:20]: ---------------------
[12:12:20]: Login to App Store Connect ({redacted})
[12:12:28]: Login successful
[12:12:32]: Price Tier unchanged (tier 0)
[12:12:34]: Uploading app icon...
[12:12:35]: source sequence is illegal/malformed utf-8
[12:12:35]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  Uploading metadata to iTunes Connect error: source sequence is illegal/malformed utf-8
    https://github.com/fastlane/fastlane/issues/13608 [open] 5 💬
    2 minutes ago

➡️  [spaceship] Do not attempt to log non-JSON bodies
    https://github.com/fastlane/fastlane/pull/13823 [open] 0 💬
    54 minutes ago

➡️  Could not upload ipa by fastlane deliver.
    https://github.com/fastlane/fastlane/issues/13815 [open] 4 💬
    5 minutes ago

and 3 more at: https://github.com/fastlane/fastlane/search?q=source%20sequence%20is%20illegal%2Fmalformed%20utf-8&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
Traceback (most recent call last):
	41: from {redacted_path}.rbenv/versions/2.5.1/bin/fastlane:23:in `<main>'
	40: from {redacted_path}.rbenv/versions/2.5.1/bin/fastlane:23:in `load'
	39: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/bin/fastlane:23:in `<top (required)>'
	38: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	37: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	36: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	35: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	34: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	33: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	32: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	31: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	30: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	29: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	28: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	27: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	26: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	25: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	24: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	23: from ../../../../../../../../../var/folders/q3/pv4n6xx50y189qbvp34h__vc0000gn/T/fl_clone20181205-74588-1hoqrir/sdc-fastfiles/env_lanes:675:in `block (2 levels) in parsing_binding'
	22: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
	21: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'
	20: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'
	19: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	18: from ../../../../../../../../../var/folders/q3/pv4n6xx50y189qbvp34h__vc0000gn/T/fl_clone20181205-74588-1hoqrir/sdc-fastfiles/env_lanes:489:in `block (2 levels) in parsing_binding'
	17: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
	16: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	15: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	14: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	13: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	12: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	11: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	10: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/fastlane/lib/fastlane/actions/upload_to_app_store.rb:16:in `run'
	 9: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/deliver/lib/deliver/runner.rb:38:in `run'
	 8: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/deliver/lib/deliver/runner.rb:123:in `upload_metadata'
	 7: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/deliver/lib/deliver/upload_assets.rb:16:in `upload'
	 6: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/tunes/app_version.rb:446:in `upload_large_icon!'
	 5: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/tunes/tunes_client.rb:774:in `upload_large_icon'
	 4: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/du/du_client.rb:29:in `upload_large_icon'
	 3: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/du/du_client.rb:75:in `upload_file'
	 2: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/client.rb:658:in `request'
	 1: from {redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/client.rb:783:in `log_request'
{redacted_path}.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.109.0/spaceship/lib/spaceship/client.rb:783:in `to_json': [!] source sequence is illegal/malformed utf-8 (JSON::GeneratorError)
</pre>
</details>

**Comment by janpio** [comment]
If you encountered this problem after updating to 2.109.0:

> PR fixing this issue is at #13825.
> 
> We are going to review and merge this as fast as possible and release 2.109.1.
> 
> Sorry for the trouble, I am ashamed (and confused) that I didn't catch this while testing the new logging code. Shouldn't have happened 🙇‍♀️

More information here: https://github.com/fastlane/fastlane/issues/13815

**Comment by SEnglishSMP** [comment]
This is still an issue with fastlane 2.112.0, even though some related bugs have been fixed.

**Comment by SEnglishSMP** [comment]
fastlane deliver --force -u *** -a *** -i /Users/vsts/agent/2.144.0/work/r1/a/_iOS 

release/drop/output/iphoneos/Distribution/_XcodeTaskExport_SMPNew/SMPNew.ipa --skip_binary_upload true -m /Users/vsts/agent/2.144.0/work/r1/a/_iOS release/drop/AppStore/metadata --skip_screenshots true --submit_for_review true
[20:57:24]: ‌Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile‌
[20:57:24]: Sending anonymous analytics information
[20:57:24]: Learn more at https://docs.fastlane.tools/#metrics
[20:57:24]: No personal or sensitive data is sent.
[20:57:24]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[20:57:24]: Login to App Store Connect (***)
[20:57:26]: Login successful

+--------------------------------------+-----------------------------------------------------------------------------------------------------------------------+
|                                                                   deliver 2.112.0 Summary                                                                    |
+--------------------------------------+-----------------------------------------------------------------------------------------------------------------------+
| force                                | true                                                                                                                  |
| username                             | ***                                                                                                     |
| app_identifier                       | ***                                                                                                   |
| ipa                                  | /Users/vsts/agent/2.144.0/work/r1/a/_iOS release/drop/output/iphoneos/Distribution/_XcodeTaskExport_SMPNew/SMPNew.ipa |
| skip_binary_upload                   | true                                                                                                                  |
| metadata_path                        | /Users/vsts/agent/2.144.0/work/r1/a/_iOS release/drop/AppStore/metadata                                               |
| skip_screenshots                     | true                                                                                                                  |
| submit_for_review                    | true                                                                                                                  |
| screenshots_path                     | ./screenshots                                                                                                         |
| app_version                          | 1.7.11115                                                                                                             |
| platform                             | ios                                                                                                                   |
| edit_live                            | false                                                                                                                 |
| use_live_version                     | false                                                                                                                 |
| skip_metadata                        | false                                                                                                                 |
| skip_app_version_update              | false                                                                                                                 |
| overwrite_screenshots                | false                                                                                                                 |
| reject_if_possible                   | false                                                                                                                 |
| automatic_release                    | false                                                                                                                 |
| phased_release                       | false                                                                                                                 |
| run_precheck_before_submit           | true                                                                                                                  |
| precheck_default_rule_level          | warn                                                                                                                  |
| ignore_language_directory_validation | false                                                                                                                 |
| precheck_include_in_app_purchases    | true                                                                                                                  |
+--------------------------------------+-----------------------------------------------------------------------------------------------------------------------+

[20:57:27]: Making sure the latest version on App Store Connect matches '1.7.11115' from the ipa file...
[20:57:29]: Successfully set the version to '1.7.11115'
[20:57:30]: Loading '/Users/vsts/agent/2.144.0/work/r1/a/_iOS release/drop/AppStore/metadata/en-US/description.txt'...
[20:57:30]: Loading '/Users/vsts/agent/2.144.0/work/r1/a/_iOS release/drop/AppStore/metadata/en-US/release_notes.txt'...
[20:57:30]: Loading '/Users/vsts/agent/2.144.0/work/r1/a/_iOS release/drop/AppStore/metadata/en-US/support_url.txt'...
[20:57:30]: Loading '/Users/vsts/agent/2.144.0/work/r1/a/_iOS release/drop/AppStore/metadata/copyright.txt'...
[20:57:31]: Activating language en-US...
[20:57:33]: Uploading metadata to App Store Connect

Looking for related GitHub issues on fastlane/fastlane...

➡️  [2.107.0] Uploading metadata to iTunes Connect error: source sequence is illegal/malformed utf-8
    https://github.com/fastlane/fastlane/issues/13608 [open] 7 💬
    4 weeks ago

➡️  [2.109.0] Could not upload with fastlane deliver: source sequence is illegal / malformed utf-8 (JSON :: GeneratorError)
    https://github.com/fastlane/fastlane/issues/13815 [closed] 9 💬
    4 weeks ago

➡️  Uploading metadata to iTunes Connect error: source sequence is illegal/malformed utf-8 
    https://github.com/fastlane/fastlane/issues/13125 [closed] 7 💬
    4 weeks ago

and 3 more at: https://github.com/fastlane/fastlane/search?q=source%20sequence%20is%20illegal%2Fmalformed%20utf-8&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/base.rb:66:in `to_json': [!] source sequence is illegal/malformed utf-8 (JSON::GeneratorError)
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/base.rb:66:in `to_json'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/tunes/tunes_client.rb:435:in `block (2 levels) in update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/client.rb:809:in `extract_key_from_block'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/client.rb:767:in `log_request'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/client.rb:658:in `request'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/tunes/tunes_client.rb:433:in `block in update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/tunes/tunes_client.rb:1436:in `with_tunes_retry'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/tunes/tunes_client.rb:432:in `update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/spaceship/lib/spaceship/tunes/app_version.rb:403:in `save!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/deliver/lib/deliver/upload_metadata.rb:130:in `upload'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/deliver/lib/deliver/runner.rb:120:in `upload_metadata'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/deliver/lib/deliver/runner.rb:38:in `run'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/deliver/lib/deliver/commands_generator.rb:79:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/deliver/lib/deliver/commands_generator.rb:182:in `run'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/deliver/lib/deliver/commands_generator.rb:17:in `start'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/fastlane/lib/fastlane/cli_tools_distributor.rb:100:in `take_off'
	from /Users/vsts/.gem-cache/gems/fastlane-2.112.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/lib/ruby/gems/2.5.0/bin/fastlane:23:in `load'
	from /usr/local/lib/ruby/gems/2.5.0/bin/fastlane:23:in `<main>'

**Comment by janpio** [comment]
Can you have a look at the matching spaceship logs and maybe post them? (`/tmp/spaceship-*.log`)

**Comment by SEnglishSMP** [comment]
@janpio Unfortunately, no.  We're using a Visual Studio Team Services hosted build server. It uses fastlane as part of its iOS publishing task.  Once the build ends, it destroys the virtual machine, so I can't go back and get the log.

**Comment by janpio** [comment]
Damn. If you have this consistently and need to get it fixed, you could probably add something to your CI scripts to output the spaceship log file to the build output. I really have no idea what could be causing this right now.

**Comment by janpio** [comment]
Yep, seems I gave you the wrong format. Recent example: `spaceship1547571768_14372_23296560.log` (Sorry!)

**Comment by SEnglishSMP** [comment]
OK, I can get the logs, but we're now being tripped up by #14049 which is occurring earlier in the process. :sob:

**Comment by SEnglishSMP** [comment]
[fastlane output.txt](https://github.com/fastlane/fastlane/files/2877569/fastlane.output.txt)
[spaceship1550526953_674_70334925750340.log](https://github.com/fastlane/fastlane/files/2877570/spaceship1550526953_674_70334925750340.log)
I'm still running into this problem with 2.116.0.  Attached are both the fastlane output and the spaceship log.  As you can see there are no errors in the spaceship log.

**Comment by SEnglishSMP** [comment]
Still getting this with fastlane 2.117.1.

**Comment by janpio** [comment]
Unfortunately I can't reproduce or even think of what could be causing this:

```
/Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/base.rb:66:in `to_json': [!] source sequence is illegal/malformed utf-8 (JSON::GeneratorError)
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/base.rb:66:in `to_json'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/tunes/tunes_client.rb:436:in `block (2 levels) in update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/client.rb:827:in `extract_key_from_block'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/client.rb:785:in `log_request'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/client.rb:676:in `request'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/tunes/tunes_client.rb:434:in `block in update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/tunes/tunes_client.rb:1455:in `with_tunes_retry'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/tunes/tunes_client.rb:433:in `update_app_version!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/spaceship/lib/spaceship/tunes/app_version.rb:403:in `save!'
	from /Users/vsts/.gem-cache/gems/fastlane-2.116.0/deliver/lib/deliver/upload_metadata.rb:135:in `upload'
```

This tells us this happens when saving the updated app, as it tried to write the request to the logfile.

One way to further debug this would be to create a Charles session file of a failing upload:
https://github.com/fastlane/fastlane/blob/master/spaceship/docs/Debugging.md#proxy-support
https://github.com/fastlane/fastlane/blob/master/ToolsAndDebugging.md#debugging-and-patching-spaceship-problems
That should probably include the "problematic" request and allow me/us to look if we can identify what trips up the logging function. (You can send it to piotrowski@gmail.com and ping me here to trigger me having a look at it.)

**Comment by SEnglishSMP** [comment]
@janpio, since we're using an Azure server, we can't set up Charles.  However, I'm not sure how this would help.  If I understand correctly, Charles is a web proxy that will allow us to capture the HTTPS traffic between fastlane and Apple.  In the log files I submitted, however, there were no errors being returned from Apple.  It appears to me that the problem is with data that is being _prepared_ to be sent to Apple.  An error occurs when attempting to _encode_ the source data, so the problem data never even gets sent to Apple.  As such, Charles won't see it.

Improved error reporting would definitely help.  If fastlane told us _what data_  had "illegal/malformed utf-8" then we would know where to look.

**Comment by janpio** [comment]
I would have tried to deduct from the previous requests where exactly we are in the process, just to pin it down - but you are not wrong with your analysis, that wouldn't help to uncover the problematic string.

Some error handling, maybe even specific to `JSON::GeneratorError` on that `to_json` might indeed be a solution. I have no idea how to output the hash in a way that would let us look at the data in a way that can help us understand it. (My first thing to try would have been to call `to_json` on it, which clearly would not have ended well). 

If you have an idea, I would be happy to merge a PR for `client.rb`, `#log_request` or `#extract_key_from_block` that `rescue`s `JSON::GeneratorError` (which is btw part of the block that is passed in to be executed - took me some time to understand that stacktrace and code).

**Comment by SEnglishSMP** [comment]
@janpio I don't know ruby, so I couldn't put together a PR for this.  But I can see from the stack trace that the failing `to_json` call is on the `data` parameter that was passed to `tunes_client.update_app_version!`  Could you add error handling to that method to log the problem `data` parameter?

**Comment by lacostej** [comment]
I made a small PR that could help troubleshooting the issue. It requires verbose to be on but it seems that you can add extra fastlane arguments to VSTS.

Some other ideas. It seems the VSTS fastlane integration is done here https://github.com/Microsoft/app-store-vsts-extension. I would
* ask whether it is possible to define a `Gemfile` or similar mechanism to point to a forked fastlane version. This would allow you to troubleshoot faster e.g. by using my proposed branch
* ask whether they can help troubleshooting the issue in some way.

**Comment by janpio** [comment]
❤️ @lacostej

**Comment by lacostej** [comment]
@SEnglishSMP version 2.122 contains troubleshooting tools. Run fastlane in verbose mode and add the logs here to see what is going on). Thanks

**Comment by SEnglishSMP** [comment]
@lacostej @janpio That you for making the code updates and merged the PR.  I will try that. We don't submit things for Apple review (which is what is failing) super often, so it may be a few weeks before I can give you the results.

**Comment by SEnglishSMP** [comment]
@lacostej @janpio Attached is the output from the new fastlane build showing the data that's causing the error.  I've replaced sensitive information with asterisks, but all of that was just normal, printable ASCII characters.

[fastlane debug output.txt](https://github.com/fastlane/fastlane/files/3179019/fastlane.debug.output.txt)

**Comment by janpio** [comment]
- No invisible characters present per https://www.soscisurvey.de/tools/view-chars.php
- https://pteo.paranoiaworks.mobi/diacriticsremover/ shows the use of `…` in the review notes
- Using http://filetools.my-addr.com/remove-non-ascii-characters-online.php also removed the `…` but also some other "special" characters in the review notes, screenshot of a diff: 
![image](https://user-images.githubusercontent.com/183673/58014391-03296280-7af9-11e9-8639-918afde55e82.png)

Can you check if the problem is gone if you remove those?

If yes, this points to our logging strategy to be non that great for non ASCII text strings :/

**Comment by SEnglishSMP** [comment]
Thanks.  Yes, I will look at removing or replacing those characters and let you know the results.  It will probably be a few weeks before we submit another build for Apple's review, though.

While I don't think removing or replacing the characters will be a problem for us, fastlane _should_ support non-ASCII characters.  Apple does.  In fact, I originally got the text by copying what we already had on iTunes Connect before we began using fastlane.  And I would imagine that other users of fastlane would want support for alternate languages with non-ASCII characters.  In short, fastlane should handle _any_ valid UTF-8 text, and if it's a problem for Apple, let Apple return an error.

**Comment by janpio** [comment]
Absolutely, we are still in trying to pinpoint the problem here. If this indeed fixes the problem, we will have to find a solution to this.

**Comment by SEnglishSMP** [comment]
Haven't yet pushed anything new to Apple review yet.  I'll update the ticket when we do.

**Comment by SEnglishSMP** [comment]
I removed the non-ASCII characters from the review notes and we're still getting the same error.  

It's worth pointing out that we don't define any review notes in the metadata we give to deliver.  I had to change the review notes for the previous build on appstoreconnect.apple.com before attempting to have deliver submit the app for review.  The spaceship logs do show the updated review notes without any non-ASCII characters, so I'm not sure why it's still having a problem.

**Comment by janpio** [comment]
Ugh, there goes our nice theory on what was going on :/
Can you post the output again so I can have another look?

**Comment by SEnglishSMP** [comment]
@janpio, here's the output from fastlane along with the debug file from spaceship.  I've replaced sensitive information with asterisks, but all of that was just normal, printable ASCII characters.

[fastlane output.txt](https://github.com/fastlane/fastlane/files/3424581/fastlane.output.txt)
[spaceship debug output.txt](https://github.com/fastlane/fastlane/files/3424582/spaceship.debug.output.txt)

**Comment by SEnglishSMP** [comment]
Still waiting on a fix.

**Comment by SEnglishSMP** [comment]
@janpio, where do we go from here?

**Comment by janpio** [comment]
I am out of ideas and honestly don't know.

**Comment by SEnglishSMP** [comment]
Still an issue.

**Comment by SEnglishSMP** [comment]
Still waiting on a fix.

**Comment by SEnglishSMP** [comment]
This is still an issue for us.

**Comment by quzongyao** [comment]
Running fastlane 2.143.0, still same issue.

**Comment by quzongyao** [comment]
I reproduced this issue, and resolved myself.

It looks like accidentally, the release_notes.txt files are encoded with something other than UTF-8, and the files just contains some characters that is could not be recognized.

After re-encoding to UTF-8, it went back to normal.

**Comment by SEnglishSMP** [comment]
We'll be trying the workaround proposed by quzongyao soon.

**Comment by SEnglishSMP** [comment]
After re-encoding all our metadata files to UTF-8, we no longer encounter this bug.
